csv/src/main/java/org/motechproject/nms/csv/service/impl/CsvAuditServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package org.motechproject.nms.csv.service.impl;
+
+import org.motechproject.nms.csv.domain.CsvAuditRecord;
+import org.motechproject.nms.csv.repository.CsvAuditRecordDataService;
+import org.motechproject.nms.csv.service.CsvAuditService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service("csvAuditService")
+public class CsvAuditServiceImpl implements CsvAuditService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsvAuditServiceImpl.class);
+    public static final String SUCCESS = "Success";
+    public static final String FAILURE = "Failure: ";
+
+    private CsvAuditRecordDataService csvAuditRecordDataService;
+
+    @Autowired
+    public void setCsvAuditRecordDataService(CsvAuditRecordDataService csvAuditRecordDataService) {
+        this.csvAuditRecordDataService = csvAuditRecordDataService;
+    }
+
+    private void audit(String file, String endpoint, String outcome) {
+        String truncatedOutcome;
+
+        if (outcome.length() > CsvAuditRecord.MAX_OUTCOME_LENGTH) {
+            truncatedOutcome = outcome.substring(0, CsvAuditRecord.MAX_OUTCOME_LENGTH);
+            LOGGER.warn("The provided outcome field was more than {} characters and was truncated",
```

Comment:
Oh, I do understand. Yep, I'll do that. & the logError stuff too. And then I'll merge, ok?